### PR TITLE
Check extension at hooks

### PR DESCRIPTION
**Pull request by aykut-bozkurt**
It might be confusing for some users that hooks are still running (since at shared_preload_libraries) even if the extension is dropped. Hence, we decided to process parquet copy hooks only when `pg_parquet` extension is created.

### CODE REVIEW

**Review by marcoslot — COMMENTED** [review]
When I create the extension from the current session, this works, but when I rerun it seems to not detect the extension.
```sql
\dx
                         List of installed extensions
┌────────────┬─────────┬────────────┬────────────────────────────────────────┐
│    Name    │ Version │   Schema   │              Description               │
├────────────┼─────────┼────────────┼────────────────────────────────────────┤
│ pg_parquet │ 0.1.0   │ public     │ copy data between Postgres and Parquet │
│ plpgsql    │ 1.0     │ pg_catalog │ PL/pgSQL procedural language           │
└────────────┴─────────┴────────────┴────────────────────────────────────────┘
(2 rows)

pg_parquet=# copy (select 1 as on) to 's3://marco-crunchy-data/tmp/one.parquet';
ERROR:  relative path not allowed for COPY to file
```

**Review by marcoslot — APPROVED** [review]

File: src/parquet_copy_hook/copy_utils.rs [hunk 1]
```diff
@@ -346,7 +365,13 @@ pub(crate) fn is_copy_from_parquet_stmt(p_stmt: &PgBox<PlannedStmt>) -> bool {
...
     let uri = copy_stmt_uri(p_stmt).expect("uri is None");
 
-    is_parquet_format_option(p_stmt) || is_parquet_uri(uri)
+    if !is_parquet_format_option(p_stmt) && !is_parquet_uri(uri) {
+        return false;
+    }
+
+    // extension checks are done via catalog (not yet searched via cache by postgres till pg18)
+    // this is why we check them after the uri checks
+    extension_exists("pg_parquet") && !extension_exists("crunchy_query_engine")
```
[the remark this answers is not in the document]
Comment (aykut-bozkurt) [reply src/parquet_copy_hook/copy_utils.rs]: makes sense

File: src/parquet_copy_hook/copy_utils.rs [hunk 2]
```diff
@@ -334,44 +334,35 @@ pub(crate) fn is_copy_to_parquet_stmt(p_stmt: &PgBox<PlannedStmt>) -> bool {
...
-
-pub(crate) fn is_copy_from_parquet_stmt(p_stmt: &PgBox<PlannedStmt>) -> bool {
-    // the GUC pg_parquet.enable_copy_hook must be set to true
-    if !ENABLE_PARQUET_COPY_HOOK.get() {
-        return false;
-    }
 
-    let is_copy_stmt = unsafe { is_a(p_stmt.utilityStmt, T_CopyStmt) };
+    // pg_parquet should be created
+    if !extension_exists("pg_parquet") {
```
Comment (marcoslot) [inline src/parquet_copy_hook/copy_utils.rs:339]: should be the last check

File: src/parquet_copy_hook/copy_utils.rs [hunk 3]
```diff
@@ -334,44 +334,35 @@ pub(crate) fn is_copy_to_parquet_stmt(p_stmt: &PgBox<PlannedStmt>) -> bool {
...
 
-    if !is_copy_stmt {
         return false;
     }
 
-    let copy_stmt = unsafe { PgBox::<CopyStmt>::from_pg(p_stmt.utilityStmt as _) };
-
-    if !copy_stmt.is_from {
+    // crunchy_query_engine should not be created
+    if extension_exists("crunchy_query_engine") {
```
Comment (marcoslot) [inline src/parquet_copy_hook/copy_utils.rs:351]: if this is the case, we want pg_parquet to be silent (should come before)

**Pull request merged**